### ISSUE TITLE
`read.dbc` falhando no Windows

### ISSUE THREAD

**Issue by filipemsc** [danicat/read.dbc#17]
Fiz alguns testes comparando Windows e Linux e parece que o pacote não funciona para arquivos muito grandes no Windows, mas roda perfeitamente no Linux.

Exemplo:
```
#install.packages("curl")
temp <- tempfile()
curl::curl_download("ftp://ftp.datasus.gov.br/dissemin/publicos/CNES/200508_/Dados/PF/PFSP2401.dbc", temp, quiet = FALSE)
read.dbc::read.dbc(temp)
```
Esse código só funciona quando estou em um WSL.

**Comment by danicat** [comment]
Eu não mexo nisso já faz muito tempo, mas verifique por favor se a package
curl faz download em modo binário. Em windows o modo binario não é default
usando a funçao *download.file* e você precisa especificar o modo binario
explicitamente. Se a função curl_download usar a download.file por debaixo
dos panos isso pode ser a causa de um arquivo corrompido.


Em qua., 30 de out. de 2024 às 14:19, Filipe Cavalcanti <
***@***.***> escreveu:

**Comment by danicat** [comment]
Por exemplo:

curl_download(url, destfile, quiet = TRUE, mode = "wb", handle = new_handle())

Experimente passar *mode = "wb"*.

Documentação: https://rdrr.io/cran/curl/man/curl_download.html

Em qua., 30 de out. de 2024 às 14:24, Daniela Petruzalek <
***@***.***> escreveu:

**Comment by filipemsc** [comment]
Sim! Isso é bem importante, mas o wb é o formato padrão do curl ```curl_download(url, destfile, quiet = TRUE, mode = "wb", handle = new_handle())```

Parece ser realmente um problema de tamanho de arquivo. O pacote funciona perfeitamente pra ler o arquivo de Alagoas, por exemplo. O problema é arquivos muito grandes do CNES, como São Paulo.

**Comment by danicat** [comment]
O que acontece se vc fiz o download no Windows e abrir no WSL e vice versa?
